multiview/embed/kcca.py

```diff
@@ -0,0 +1,238 @@
+"""
+kcca.py
+====================================
+Python module for kernel canonical correlation analysis (kCCA)
+
+Adopted from Steven Van Vaerenbergh's MATLAB Package 'KMBOX'
+https://github.com/steven2358/kmbox
+"""
+
+from .base import BaseEmbed
+from ..utils.utils import check_Xs
+
+import numpy as np
+import numpy.matlib
+from scipy import linalg
+
+class KCCA(BaseEmbed):
+    """
+    CCA aims to find useful projections of the
+    high- dimensional variable sets onto the compact
+    linear representations, the canonical components (components_).
+
+    Each resulting canonical variate is computed from
+    the weighted sum of every original variable indicated
+    by the canonical weights (weights_).
+
+    The canonical correlation (cancorrs_) quantifies the linear
+    correspondence between the two views of data
+    based on Pearson’s correlation between their
+    canonical components.
+
+    Canonical correlation can be seen as a metric of
+    successful joint information reduction between two views
+    and, therefore, routinely serves as a performance measure for CCA.
+
+    The kernel generalization of CCA, kernel CCA, is used when
+    there are nonlinear relations between two views.
+
+    Parameters
+    ----------
+    reg : float, default = 0.1
+          Regularization parameter
+    n_components : int, default = 10
+                   Number of canonical dimensions to keep
+    ktype : string, default = 'linear'
+            Type of kernel
+        - value can be 'linear', 'gaussian' or 'polynomial'
+    cutoff : float, default = 1x10^-15
+             Optional regularization parameter
+             to perform spectral cutoff when computing the canonical
+             weight pseudoinverse during held-out data prediction
+    sigma : float, default = 1.0
+            Parameter if Gaussian kernel
+    degree : integer, default = 2
+             Parameter if Polynomial kernel
+
+    """
+    
+    def __init__(
+        self,
+        reg=0.00001,
+        n_components=10,
```

Comment:
Seems high? Idk if will throw an error if < 10 components possible

Reply:
not sure what you mean here. do you want me to set it to like 2?

Reply:
set it to 2

Reply:
yeah, i think 2 is a default good number and shouldn't be too large for an error